contracts/sdk/AQueryProofVerifierBuilder.sol

```diff
@@ -0,0 +1,167 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.22;
+
+import {Strings} from "@openzeppelin/contracts/utils/Strings.sol";
+import {Initializable} from "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+
+import {PublicSignalsBuilder} from "./PublicSignalsBuilder.sol";
+
+import {INoirVerifier} from "../interfaces/verifiers/INoirVerifier.sol";
+
+/**
+ * @title Abstract Query Proof Verifier Builder
+ * @notice An abstract contract providing a framework for verifying ZK proofs related to user queries,
+ * supporting both Circom (Groth16) and Noir systems.
+ */
+abstract contract AQueryProofVerifierBuilder is Initializable {
+    using Strings for uint256;
+    using PublicSignalsBuilder for uint256;
+
+    uint256 public constant _PROOF_SIGNALS_COUNT = 23;
+
+    struct ABuilderStorage {
+        address registrationSMT;
+        address votingVerifier;
+    }
+
+    struct ProofPoints {
+        uint256[2] a;
+        uint256[2][2] b;
+        uint256[2] c;
+    }
+
+    error FailedToCallVerifyProof();
+    error InvalidNoirProof(bytes32[] pubSignals, bytes zkPoints);
+    error InvalidCircomProof(uint256[] pubSignals, ProofPoints zkPoints);
+
+    function __AQueryProofVerifierBuilder_init(
+        address registrationSMT_,
+        address votingVerifier
```

Comment:
Let's have a `public view function` to read the verifier + an `internal function` to set it. (only the verifier).